pkg/test/utils/kubernetes.go

```diff
@@ -1321,3 +1398,29 @@ func Kubeconfig(cfg *rest.Config, w io.Writer) error {
 		},
 	}, w)
 }
+
+func constructGVK(apiVersion, kind string) schema.GroupVersionKind {
```

Comment:
This is a nice helper method, but it would be even more powerful it you
- changed it from a bare `func` to a method of `TestResourceRef`
- made it also internally create an `*Unstructured`:
  ```
		gvk := constructGVK(resourceRef.ApiVersion, resourceRef.Kind)
		referencedResource := &unstructured.Unstructured{}
		referencedResource.SetGroupVersionKind(gvk)
  ```
- and called it `NewUnstructured` or similar.

Thinking about it, it might make sense to have it also return a `NamespacedName` and an `error` in case crucial fields (like name and kind) are missing. On one hand, if we don't do any validation here, then the client would probably catch such garbage later. But on the other hand, we could use such validation function earlier, just after loading the assert files, in order to give user feedback on critical errors ASAP.

When you're at it, please also add a couple of test cases for it.

Reply:
`NamespacedName`/`*unstructured.Unstructured` would be required when we evaluate the expressions. If the goal is also to validate, we should have two separate methods:
```go
func (t *TestResourceRef) BuildResourceReference() (namespacedName types.NamespacedName, referencedResource *unstructured.Unstructured)

func (t *TestResourceRef) Validate() (error)
```